local_pathfinding/obstacles.py

```diff
@@ -0,0 +1,254 @@
+
+        # contains() requires a shapely Point object as an argument
+        point = Point(*point)
+
+        return not self.collision_zone.contains(point)
+
+    def update_collision_zone(self, collision_zone: Polygon):
+        """Updates the collision zone of the obstacle. Called by the child classes.
+
+        Args:
+            collision_zone (Polygon): Shapely Polygon representing the obstacle's collision zone.
+        """
+        self.collision_zone = collision_zone
+
+    def update_sailbot_data(self, sailbot_position: LatLon, sailbot_speed: float):
+        """Updates the sailbot's position and speed.
+
+        Args:
+            sailbot_position (LatLon): LatLon position of the SailBot.
+            sailbot_speed (float): Speed of the SailBot in kmph.
+        """
+        self.sailbot_position_latlon = sailbot_position
+        self.sailbot_position = latlon_to_xy(self.reference, sailbot_position)
+        self.sailbot_speed = sailbot_speed
+
+    def update_reference_point(self, reference: LatLon):
+        """Updates the reference point.
+
+        Args:
+            reference (LatLon): LatLon position of the updated global waypoint.
+        """
+        self.reference = reference
+        self.sailbot_position = latlon_to_xy(self.reference, self.sailbot_position_latlon)
+
+        if isinstance(self, Boat):
+            # regenerate collision zone with updated reference point
+            self.update_collision_zone(self.create_boat_collision_zone(self.ais_ship))
+
+    @staticmethod
+    def create_collision_zone(collision_zone_poly, centre_position: XY) -> Polygon:
+        """Creates a final Shapely Polygon which represents the collision zone of the obstacle.
+
+        Args:
+            collision_zone_poly (Polygon): Polygon which defines the shape of the collision zone.
+            centre_position (XY): XY position of the centre of the collision zone.
+
+        Returns:
+            Polygon: Buffered and translated collision zone polygon.
+        """
+        collision_zone_poly = af.translate(collision_zone_poly, *centre_position)
+        return collision_zone_poly.buffer(COLLISION_ZONE_SAFETY_BUFFER, join_style=2)
+
+
+class Boat(Obstacle):
+    """Describes boat objects which Sailbot must avoid.
+    Also referred to target ships or boat obstacles.
+
+    Attributes:
+        id (int): MMSI ID of the boat.
+        width (float): Width of the boat in meters.
+        length (float): Length of the boat in meters.
+    """
+
+    def __init__(
+        self,
+        reference: LatLon,
+        sailbot_position: LatLon,
+        sailbot_speed: float,
+        ais_ship: HelperAISShip,
+    ):
+        super().__init__(reference, sailbot_position, sailbot_speed)
+
+        self.ais_ship = ais_ship
+        self.update_collision_zone(self.create_boat_collision_zone(self.ais_ship))
+
+    def create_boat_collision_zone(self, ais_ship: HelperAISShip) -> Polygon:
```

Comment:
How about we make this `update_boat_collision_zone`? Then we can call it directly in init. The last line of this function would call `self.update_collision_zone`